src/HTML5/Parser/Tokenizer.php

```diff
@@ -121,16 +121,30 @@ public function setTextMode($textmode, $untilTag = null)
      */
     protected function consumeData()
     {
-        // Character Ref
-        /*
-         * $this->characterReference() || $this->tagOpen() || $this->eof() || $this->characterData();
-         */
+        // Character reference
         $this->characterReference();
-        $this->tagOpen();
-        $this->eof();
 
-        // Inline the parsing of characters as it's the critical performance path
+        // Parse tag
+        if ($this->scanner->current() === '<') {
```

Comment:
You could move the `$tok = $this->scanner->current();` from line 144 to before the `if`-Statement to remove another `$this->scanner->current()` call.

Reply:
I don't think I can because the content of the if change the scanner position, so I need to get the current one after it. I may be wrong.

Reply:
I don't think this is a problem, as line 132 (`$tok = $this->scanner->next();`) inside the `if` overrides the previously set `$tok` anyway.

Reply:
Have a look inside the "or calls" : they do change the current position of the cursor :)

Reply:
Ah:) 

Still, you could move the call before and add another one to pick up the current scanner state at the end of the `if` clause. Then you'd have the same number of function calls for the "begin tag" case, but one less for all other cases.

Reply:
Great idea indeed! I tried and it does improve even more the performance :) .